spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/listener/errorhandler/ExponentialBackoffErrorHandler.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2013-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sqs.listener.errorhandler;
+
+import io.awspring.cloud.sqs.MessageHeaderUtils;
+import io.awspring.cloud.sqs.listener.QueueMessageVisibility;
+import io.awspring.cloud.sqs.listener.SqsHeaders;
+import io.awspring.cloud.sqs.listener.Visibility;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * An implementation of an Exponential Backoff error handler for asynchronous message processing.
+ *
+ * <p>
+ * This error handler sets the SQS message visibility timeout exponentially
+ * based on the number of received attempts whenever an exception occurs.
+ *
+ * <p>
+ * When AcknowledgementMode is set to ON_SUCCESS (the default),
+ * returning a failed future prevents the message from being acknowledged.
+ *
+ * @author Bruno Garcia
+ * @author Rafael Pavarini
+ */
+
+public class ExponentialBackoffErrorHandler<T> implements AsyncErrorHandler<T> {
+	/**
+	 * The default initial visibility timeout interval.
+	 */
+	public static final int DEFAULT_INITIAL_VISIBILITY_TIMEOUT_SECONDS = 100;
+	/**
+	 * The default multiplier, which doubles the visibility timeout.
+	 */
+	public static final double DEFAULT_MULTIPLIER = 2.0;
+	/**
+	 * The default maximum visibility timeout interval,
+	 * which corresponds to the maximum SQS visibility timeout of 12 hours.
+	 */
+	public static final int DEFAULT_MAX_VISIBILITY_TIMEOUT_SECONDS = 43200;
+
+	private final int initialVisibilityTimeoutSeconds;
+	private final double multiplier;
+	private final int maxVisibilityTimeoutSeconds;
+
+	public ExponentialBackoffErrorHandler() {
+		this(DEFAULT_INITIAL_VISIBILITY_TIMEOUT_SECONDS, DEFAULT_MULTIPLIER, DEFAULT_MAX_VISIBILITY_TIMEOUT_SECONDS);
+	}
+
+	public ExponentialBackoffErrorHandler(int initialVisibilityTimeoutSeconds, double multiplier) {
+		this(initialVisibilityTimeoutSeconds, multiplier, DEFAULT_MAX_VISIBILITY_TIMEOUT_SECONDS);
+	}
+
+	public ExponentialBackoffErrorHandler(int initialVisibilityTimeoutSeconds, double multiplier, int maxVisibilityTimeoutSeconds) {
```

Comment:
How about instead of multiple constructors we create an `ExponentialBackoffErrorHandlerBuilder`?